app/scripts/modules/core/src/entityTag/EntityTagEditor.tsx

```diff
@@ -0,0 +1,322 @@
+import { IDeferred } from 'angular';
+import { $q } from 'ngimport';
+import * as Formsy from 'formsy-react';
+import * as marked from 'marked';
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { IModalServiceInstance } from 'angular-ui-bootstrap';
+import autoBindMethods from 'class-autobind-decorator';
+
+import {
+  UUIDGenerator, Application, EntityTagWriter, HelpField, IEntityRef, IEntityTag,
+  ReactInjector, TaskMonitor, TaskMonitorBuilder
+} from 'core';
+
+import { BasicFieldLayout, TextArea, ReactModal } from 'core/presentation';
+import { NgReact } from 'core/reactShims/ngReact';
+import { EntityRefBuilder } from './entityRef.builder';
+
+import './entityTagEditor.modal.less';
+
+export interface IOwnerOption {
+  label: string;
+  type: string;
+  owner: any;
+  isDefault: boolean;
+}
+
+export interface IEntityTagEditorProps {
+  owner: any;
+  application: Application;
+  entityType: string;
+  tag: IEntityTag;
+  ownerOptions: IOwnerOption[];
+  entityRef: IEntityRef;
+  isNew: boolean;
+  show?: boolean;
+  onHide?(event: any): void;
```

Comment:
The caller can access the trigger event from the click event handler or whatever.  IDK if it's necessary.

I used `onHide.apply` to propagate any arguments